### DISCUSSION TITLE
DataGrid Row with Button

### DISCUSSION THREAD

**Discussion by drei97** [AvaloniaUI/Avalonia#16845] [Q&A] [answered]
Hi,

I have a DataGrid and in one column there is a Button. When the user is pressing that button some action should take place dealing with the data of that row.

How to detect from which row the button was pressed? A row of a DataGrid has several states, "selected" or "highlighted". When I move the mouse to a row which is not selected it's highlighted. When now the button is pressed the command method (ExpandItem) is invoked before the new row is set as "selected". The property SelectedItem is still the old one.

View:
```
<DataGridTemplateColumn Header="Column Example">
    <DataGridTemplateColumn.CellTemplate>
        <DataTemplate>
            <Button Command="{Binding DataContext.CommandClick, 
                    RelativeSource={RelativeSource FindAncestor, AncestorType={x:Type UserControl}}}"
                    CommandParameter="Expand"
            </Button>
        </DataTemplate>
    </DataGridTemplateColumn.CellTemplate>
</DataGridTemplateColumn>

```

ViewModel:

```
...
        Commands.Add("Expand", ExpandItem);
...


    private void ExpandItem()
    {
        var selected = SelectedItem; // <== ?? No, old row still selected

        if (selected is not null)
        {
            if (selected.Childs is not null)
            {
                selected.IsExpanded = !selected.IsExpanded;
                Refresh();
            }
        }
    }

```

J.

**Comment by thevortexcloud** [comment, marked as answer]
> How to detect from which row the button was pressed? 

The easiest way is to just pass the datacontext of the row to your command parameter, which will be the row that was clicked on.

**Reply by drei97** [reply]
got it!
